ide/nativeimage.api/test/unit/src/org/netbeans/modules/nativeimage/debug/TestNIDebuggerProvider.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.netbeans.modules.nativeimage.debug;
+
+import java.io.File;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Consumer;
+
+import org.netbeans.api.extexecution.ExecutionDescriptor;
+import org.netbeans.modules.nativeimage.api.debug.NIFrame;
+import org.netbeans.modules.nativeimage.api.debug.NIVariable;
+import org.netbeans.modules.nativeimage.spi.debug.NIDebuggerProvider;
+import org.netbeans.modules.nativeimage.spi.debug.filters.FrameDisplayer;
+import org.netbeans.modules.nativeimage.spi.debug.filters.VariablesDisplayer;
+
+public class TestNIDebuggerProvider implements NIDebuggerProvider {
+
+    private final Map<Object, Object> breakpoints = new HashMap<>();
+    private FrameDisplayer frameDisplayer;
+    private VariablesDisplayer variablesDisplayer;
+
+    public TestNIDebuggerProvider() {
+    }
+
+    @Override
+    public Object addLineBreakpoint(Object id, String filePath, int line, boolean enabled, String condition, boolean hidden) {
+        String testBP = "TEST " + id + filePath + line + enabled + condition + hidden;
+        breakpoints.put(id, testBP);
+        return testBP;
+    }
+
+    @Override
+    public void removeBreakpoint(Object id) {
+        breakpoints.remove(id);
+    }
+
+    @Override
+    public void setFrameDisplayer(FrameDisplayer frameDisplayer) {
+        this.frameDisplayer = frameDisplayer;
+    }
+
+    @Override
+    public void setVariablesDisplayer(VariablesDisplayer variablesDisplayer) {
+        this.variablesDisplayer = variablesDisplayer;
+    }
+
+    @Override
+    public void start(List<String> command, File workingDirectory, String debugger, String displayName, ExecutionDescriptor executionDescriptor, Consumer<Object> startedEngine, Runnable finishedCallback) {
```

Comment:
I didn't want to introduce a dependency on api.debugger just because of this. But, maybe it's not a big deal, I may add the dependency and change it to Consumer<DebuggerEngine>.

Reply:
Changed to `Consumer<DebuggerEnfgine>`.